core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java

```diff
@@ -24,7 +24,7 @@
     private final Metrics[] defaultMetrics;
 
     public MetricsPublisher(final Context context) {
-        this.context = context;
+        this.context = nonNull(context, "context");
```

Comment:
This should be `this.context = nonNull(context, "context").getApplicationContext();` for safety